activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -225,8 +225,8 @@ def pick(*keys)
   #   [1, "", nil, 2, " ", [], {}, false, true].compact_blank
   #   # =>  [1, 2, true]
   #
-  #   Set.new([nil, "", 1, 2])
-  #   # => [2, 1] (or [1, 2])
+  #   Set.new([nil, "", 1, 2]).compact_blank
+  #   # => [1, 2]
```

Comment:
We could sidestep the issue of order by including only one non-blank value:

```suggestion
  #   Set.new([nil, "", 1, false]).compact_blank
  #   # => [1]
```

Reply:
Sure don't see why not, though aren't `Hash` (and by extension `Set`) ordered anyway since Ruby 1.9?

Reply:
As an implementation detail, `Set` might be ordered, but it is ["a collection of unordered values with no duplicates"](https://docs.ruby-lang.org/en/2.7.0/Set.html).  Also note that for `to_a` ["the order of elements is uncertain"](https://docs.ruby-lang.org/en/2.7.0/Set.html#method-i-to_a).

Reply:
Ok, fair!